### PR TITLE
add breadcrumb navigation to header of project_questions

### PR DESCRIPTION
**Pull request by MyPyDavid**
this is an implementation for https://github.com/rdmorganiser/rdmo/issues/496

I left the header text "Questionnaire" and added the breadcrumb just below it in [project_questions_head.html](rdmo/projects/templates/projects/project_questions_head.html) ..

Ive added a tiny bit of js because I could not find the right argument to give to `.jump()` to jump the current section in the interview.
Inside the if-block of [project_questions/services.js](rdmo/projects/static/projects/js/project_questions/services.js) line 805 in service.jump :
```js  
// jump to first questionset of the section in breadcrumb
let section_from_service = service.project.catalog.sections.find(x => x.id === section.id)
service.initView(section_from_service.questionsets[0].id);
```
in [project_questions_head.html](rdmo/projects/templates/projects/project_questions_head.html)
```html
<a href="" ng-click="service.jump(service.questionset.section)">
<span> {$ service.questionset.section.title $} </span>
</a>
```

### CODE REVIEW

**Comment by MyPyDavid** [comment]
this PR should wait for https://github.com/rdmorganiser/rdmo/pull/512 `1.8.3` because it also changes the [project_questions/services.js](https://github.com/rdmorganiser/rdmo/pull/rdmo/projects/static/projects/js/project_questions/services.js)

File: rdmo/projects/static/projects/js/project_questions/services.js [hunk 1]
```diff
@@ -808,7 +808,13 @@ angular.module('project_questions')
                 if (angular.isDefined(questionset)) {
                     service.initView(questionset.id);
                 } else if (angular.isDefined(section)) {
-                    service.initView(section.questionsets[0].id);
+                    if (angular.isDefined(section.questionset)) {
+                        service.initView(section.questionsets[0].id);
+                    } else {
+                        // jump to first questionset of the section in breadcrumb
+                        let section_from_service = service.project.catalog.sections.find(x => x.id === section.id)
```
Comment (MyPyDavid) [inline rdmo/projects/static/projects/js/project_questions/services.js:815]: Need an angularJS version of `.find()` here?

Comment (jochenklar) [reply rdmo/projects/static/projects/js/project_questions/services.js:815]: You can use:

```
var section_from_service = $filter('filter')(service.project.catalog.sections, {
    id: section.id
})[0]
```

File: rdmo/projects/static/projects/js/project_questions/services.js [hunk 2]
```diff
@@ -817,7 +823,13 @@ angular.module('project_questions')
             if (angular.isDefined(questionset)) {
                 service.initView(questionset.id);
             } else if (angular.isDefined(section)) {
-                service.initView(section.questionsets[0].id);
+                if (angular.isDefined(section.questionset)) {
+                    service.initView(section.questionsets[0].id);
+                } else {
+                    // jump to first questionset of the section in breadcrumb
+                    let section_from_service = service.project.catalog.sections.find(x => x.id === section.id)
```
[the remark this answers is not in the document]
Comment (jochenklar) [reply rdmo/projects/static/projects/js/project_questions/services.js:830]: same as above

**Comment by jochenklar** [comment]
Hi @MyPyDavid , the functionality is ok, I don't like the layout. We could do it like in the screenshot (but with catalog and section as clickable links):

image

Important detail, the sidebar is aligned to the main part for the headline "Overview" as well as the text.

**Comment by jochenklar** [comment]
Alternatively, we use the bootstrap breadcrumps (https://getbootstrap.com/docs/3.3/components/#breadcrumbs) and get rid of the whole headline of the QuestionSet (`Contributors` in the screenshot above)

File: rdmo/projects/static/projects/js/project_questions/services.js (hunk 1 above)
Comment (MyPyDavid) [reply rdmo/projects/static/projects/js/project_questions/services.js]: this works, thanks!

**Comment by MyPyDavid** [comment]
Ive used the bootstrap breadcrumb now and kept alignment with sidebar via `margin-top : 70px`.

Any of these alternatives?

### 1
breadcrumb-1

with longer titles it comes in a newline..
### 2
breadcrumb-2
### 3
breadcrumb-3

**Comment by jochenklar** [comment]
Hey @MyPyDavid , thanks. I like the first one, but with a smaller font. I am not sure if we should keep the `Questionnaire` title or not. It seems a bit redundant, but to have some indication what is happening on the site might be good. The main format is the same on all RDMO pages:

```
h1

main        sidebar
```

I don't want to break in only on one page.

Maybe:

```
catalog / section / qeustionset (soon page)

help_text                           sidebar


```

I am also not super exited about the "well", but we should maybe ask other people.

**Comment by jochenklar** [comment]
2 more styling ideas:

```
    .breadcrumb {
        margin-top: 20px;
        margin-bottom: 12px;

        background-color: transparent;
        border: 1px solid #ccc;
    }
```

and 

```
    .breadcrumb {
        margin-top: 70px;
        margin-bottom: 0;
        background-color: transparent;

        padding-right: 0;
        padding-left: 0;
    }
```

**Comment by jochenklar** [comment]
image
image

**Comment by MyPyDavid** [comment]
I think the **border box** version is also nice! This makes the breadcrumb clearly separate from the title and fills up the big empty space in the top from `margin-top: 70px`.

**Comment by MyPyDavid** [comment]
update branch from https://github.com/rdmorganiser/rdmo/tree/pr/519-breadcrumb/rdmo

**Comment by jochenklar** [comment]
Thanks! Can be merged when the SG gives their ok.

**Comment by MyPyDavid** [comment]
- [ ]  For "My Projects" add:

```
<li>
        <a href="{% url 'projects' %}">{% trans 'My projects' %}</a>
</li>
```
rdmo/projects/templates/projects/project_questions_head.html

**Comment by MyPyDavid** [comment]
now it looks like this:
image

**Pull request merged**